components/script/script_runtime.rs

```diff
@@ -241,6 +242,17 @@ unsafe extern "C" fn get_incumbent_global(_: *const c_void, _: *mut RawJSContext
     result
 }
 
+#[allow(unsafe_code)]
+unsafe extern "C" fn run_jobs(microtask_queue: *const c_void, cx: *mut RawJSContext) {
+    let cx = JSContext::from_ptr(cx);
+    wrap_panic(&mut || {
+        let microtask_queue = &*(microtask_queue as *const MicrotaskQueue);
+        // TODO: run Promise- and User-variant Microtasks, and do #notify-about-rejected-promises.
+        // Those will require real `target_provider` and `globalscopes` values.
```

Comment:
to be honest, i’m not yet sure how to plumb those values into here across the rust-C++-rust ffi boundaries. doing so may require further mozjs changes.

Reply:
If I understand correctly, I think the easiest thing here will be to add a method to ScriptThread like this:
```rs
fn perform_a_microtask_checkpoint_with_queue(queue: Rc<MicrotaskQueue>, can_gc: CanGc) {
    with_script_thread(|script_thread| script_thread.do_perform_a_microtask_checkpoint_with_queue(queue, can_gc))
}
```
and extract `ScriptThread::perform_microtask_checkpoint` into `ScriptThread::do_perform_a_microtask_checkpoint_with_queue`, and `ScriptThread::perform_a_microtask_checkpoint` can pass `self.microtask_queue`.

Reply:
Granted, that will need adjusting when we support debugging worker globals. Then we'll need to find a way to invoke https://doc.servo.org/script/dom/globalscope/struct.GlobalScope.html#method.perform_a_microtask_checkpoint instead.